fuel-merkle/src/common/position.rs

```diff
@@ -110,43 +114,84 @@ impl Position {
 
     /// Construct a position from a leaf index. The in-order index corresponding
     /// to the leaf index will always equal the leaf index multiplied by 2.
-    pub fn from_leaf_index(index: u64) -> Self {
-        Position(index * 2)
+    /// Panics if index is too large to fit into u64.
+    pub fn from_leaf_index(index: u64) -> Option<Self> {
+        Some(Position(index.checked_mul(2)?))
+    }
+
+    #[cfg(test)]
+    pub(crate) fn from_leaf_index_unwrap(index: u64) -> Self {
+        Self::from_leaf_index(index).expect("Index too large")
     }
 
     /// The sibling position.
     /// A position shares the same parent and height as its sibling.
-    pub fn sibling(self) -> Self {
-        let shift = 1 << (self.height() + 1);
-        let index = self.in_order_index() as i64 + shift * self.direction();
-        Self::from_in_order_index(index as u64)
+    pub fn sibling(self) -> Result<Self, GetNodeError> {
+        #[allow(clippy::arithmetic_side_effects)] // height() <= 64
+        let shift = 1u64
+            .checked_shl(self.height() + 1)
+            .ok_or(GetNodeError::CannotExist)?;
+        let this = self.in_order_index();
+        Ok(Self::from_in_order_index(match self.orientation() {
+            Side::Left => this.checked_sub(shift).ok_or(GetNodeError::CannotExist)?,
+            Side::Right => this.checked_add(shift).ok_or(GetNodeError::CannotExist)?,
+        }))
     }
 
     /// The parent position.
     /// The parent position has a height less 1 relative to this position.
-    pub fn parent(self) -> Self {
-        let shift = 1 << self.height();
-        let index = self.in_order_index() as i64 + shift * self.direction();
-        Self::from_in_order_index(index as u64)
+    pub fn parent(self) -> Result<Self, GetNodeError> {
+        let shift = 1u64
+            .checked_shl(self.height())
+            .ok_or(GetNodeError::CannotExist)?;
+        let this = self.in_order_index();
+        Ok(Self::from_in_order_index(match self.orientation() {
```

Comment:
Hmm, we could also panic inside of the `self.orientation`:

<img width="772" alt="image" src="https://github.com/FuelLabs/fuel-vm/assets/18346821/c8fc30ed-9082-4e46-8d62-4ea843afaae2">